From c63c0bbf2b7e26e1085b7110558b2b340e0d2446 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 30 Jan 2024 18:36:23 +0100
Subject: [PATCH 0725/1695] drivers: crypto: stm32: clean function references

Remove useless & operator in function references of stm32 crypto drivers.
No functional changes.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 60dd8819faa7da0270c6c26adb0c134b4034820b drivers: crypto: stm32: clean function references]
(cheery picked from commit 061e13f64e84 drivers: crypto: stm32: clean function references)
Reviewed-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Change-Id: Id9a1d57a723c34fca0fc9b7992059cd44308ab06
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372863
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/drivers/crypto/stm32/authenc.c    | 18 +++++++++---------
 core/drivers/crypto/stm32/cipher.c     | 24 ++++++++++++------------
 core/drivers/crypto/stm32/stm32_saes.c |  2 +-
 3 files changed, 22 insertions(+), 22 deletions(-)

diff --git a/core/drivers/crypto/stm32/authenc.c b/core/drivers/crypto/stm32/authenc.c
index 4ff320c3f..bf305b02f 100644
--- a/core/drivers/crypto/stm32/authenc.c
+++ b/core/drivers/crypto/stm32/authenc.c
@@ -355,15 +355,15 @@ static TEE_Result stm32_ae_allocate(void **ctx, uint32_t algo)
  * Registration of the Authenc Driver
  */
 static struct drvcrypt_authenc driver_authenc = {
-	.alloc_ctx = &stm32_ae_allocate,
-	.free_ctx = &stm32_ae_free,
-	.init = &stm32_ae_initialize,
-	.update_aad = &stm32_ae_update_aad,
-	.update_payload = &stm32_ae_update_payload,
-	.enc_final = &stm32_ae_enc_final,
-	.dec_final = &stm32_ae_dec_final,
-	.final = &stm32_ae_final,
-	.copy_state = &stm32_ae_copy_state,
+	.alloc_ctx = stm32_ae_allocate,
+	.free_ctx = stm32_ae_free,
+	.init = stm32_ae_initialize,
+	.update_aad = stm32_ae_update_aad,
+	.update_payload = stm32_ae_update_payload,
+	.enc_final = stm32_ae_enc_final,
+	.dec_final = stm32_ae_dec_final,
+	.final = stm32_ae_final,
+	.copy_state = stm32_ae_copy_state,
 };
 
 TEE_Result stm32_register_authenc(void)
diff --git a/core/drivers/crypto/stm32/cipher.c b/core/drivers/crypto/stm32/cipher.c
index ec66c4f6f..8848e6f01 100644
--- a/core/drivers/crypto/stm32/cipher.c
+++ b/core/drivers/crypto/stm32/cipher.c
@@ -249,21 +249,21 @@ static TEE_Result stm32_saes_cipher_allocate(void **ctx, uint32_t algo)
 }
 
 static struct drvcrypt_cipher driver_cipher_cryp = {
-	.alloc_ctx = &stm32_cryp_cipher_allocate,
-	.free_ctx = &stm32_cipher_free,
-	.init = &stm32_cipher_initialize,
-	.update = &stm32_cipher_update,
-	.final = &stm32_cipher_final,
-	.copy_state = &stm32_cipher_copy_state,
+	.alloc_ctx = stm32_cryp_cipher_allocate,
+	.free_ctx = stm32_cipher_free,
+	.init = stm32_cipher_initialize,
+	.update = stm32_cipher_update,
+	.final = stm32_cipher_final,
+	.copy_state = stm32_cipher_copy_state,
 };
 
 static struct drvcrypt_cipher driver_cipher_saes = {
-	.alloc_ctx = &stm32_saes_cipher_allocate,
-	.free_ctx = &stm32_cipher_free,
-	.init = &stm32_cipher_initialize,
-	.update = &stm32_cipher_update,
-	.final = &stm32_cipher_final,
-	.copy_state = &stm32_cipher_copy_state,
+	.alloc_ctx = stm32_saes_cipher_allocate,
+	.free_ctx = stm32_cipher_free,
+	.init = stm32_cipher_initialize,
+	.update = stm32_cipher_update,
+	.final = stm32_cipher_final,
+	.copy_state = stm32_cipher_copy_state,
 };
 
 TEE_Result stm32_register_cipher(enum stm32_cipher_ip_id cipher_ip)
diff --git a/core/drivers/crypto/stm32/stm32_saes.c b/core/drivers/crypto/stm32/stm32_saes.c
index 66738e877..9ecca31df 100644
--- a/core/drivers/crypto/stm32/stm32_saes.c
+++ b/core/drivers/crypto/stm32/stm32_saes.c
@@ -1460,5 +1460,5 @@ static const struct dt_device_match saes_match_table[] = {
 DEFINE_DT_DRIVER(stm32_saes_dt_driver) = {
 	.name = "stm32-saes",
 	.match_table = saes_match_table,
-	.probe = &stm32_saes_probe,
+	.probe = stm32_saes_probe,
 };
-- 
2.47.2

